calculators/js/vampgrowth.init.js

```diff
@@ -109,14 +109,14 @@ $( document ).ready( function () {
 			var data, graphData, resultsOnly, timeframeSolution, text, textBeforeTime, parts,
 				isBuffy = $isBuffy.prop( 'checked' ),
 				hasImmigration = $hasImmigration.prop( 'checked' ),
-				varInitPopulation = math.eval( $initial.val() ) || 5,
-				varGrowthFactorTimeframe = math.eval( $growthFactorTimeframe.val() ) || 7,
-				varBuffyDampeningTimeframe = math.eval( $buffyFactorTimeframe.val() ) || 7,
-				varGrowthFactor = ( math.eval( $growthFactor.val() ) || 2 ) / varGrowthFactorTimeframe,
-				varBuffyDampening = ( math.eval( $buffyFactor.val() ) || 2 ) / varBuffyDampeningTimeframe,
-				varTimeframe = math.eval( $resultTimeframe.val() ),
-				varImmigration = ( math.eval( $vampImmigration.val() ) || 0 ),
-				varImmigrationTimeframe = ( math.eval( $vampImmigrationTimeframe.val() ) || 7 );
+				varInitPopulation = parseInt( $initial.val() || 5),
+				varGrowthFactorTimeframe = parseInt( $growthFactorTimeframe.val() || 7 ),
+				varBuffyDampeningTimeframe = parseInt( $buffyFactorTimeframe.val() || 7 ),
+				varGrowthFactor = ( $growthFactor.val() || 2 ) / varGrowthFactorTimeframe,
+				varBuffyDampening = ( $buffyFactor.val() || 2 ) / varBuffyDampeningTimeframe,
+				varTimeframe = parseInt( $resultTimeframe.val() ),
+				varImmigration = ( parseInt( $vampImmigration.val() || 0 ) ),
+				varImmigrationTimeframe = ( parseInt( $vampImmigrationTimeframe.val() || 7 ) );
```

Comment:
The switch from `math.eval` to `parseInt` is optional, I think. The major difference here is that we fall back onto the conditional assignment if the output of `val()` (a string) is false or undefined. The string "0" is true, while the integer value 0 is false, so this should solve incorrect reassignment away from 0!

Reply:
So, the reason I shied away from "parseInt" is because there are cases where you might want to use fractions in the calculator (see example tweet: https://twitter.com/515mm/status/1027567194614456320 ) 

I thought of using parseFloat() instead, but then that won't allow for things like "1/2" or "365/2" etc, which I thought was a nice touch to allow people to insert if they want to, but that's only possible with the mathjs extension (math.eval)

All that is to say that I think that the only real way to validate here is to literally check for empty value in the input. 

I plan to rewrite a large part of the calculation itself, but the validation should stay the same. So I'd add a validation method, something like this -

```javascript
getInputValue = function ( $jquery, fallback ) {
   return $jquery.val() !== '' ? $jquery.val() : fallback;
}
```

And then to each of the vars:

```
varBuffyDampening = math.eval( getInputValue( $buffyFactor, 2 ) / varBuffyDampeningTimeframe,
varTimeframe = math.eval( getInputValue( $resultTimeframe, 7 ) ),
varImmigration = math.eval( getInputValue( $vampImmigration, 0 ) ),
varImmigrationTimeframe = math.eval( getInputValue( $vampImmigrationTimeframe, 7 ) );
```
etc.

If you have time an inclination to fix this, go ahead! Otherwise, please don't feel obliged to, I'm going to rewrite a part of the calculation anyways so I can just fix this along the way -- and you already are the one who **found** this issue as it is, and I thank you for that!! :)

Reply:
That makes sense to me! I generally shy away from `eval`-style commands that give too much power to user input, but the flexibility you're describing is really nice, and in any case I think the exponential growth of the vampire community is even more frightening than arbitrary code execution ;)

I'd be happy to try my hand at adding a validation method, if you don't object to the intrusion! It would add a lot of legitimacy to the epitaph I want on my headstone:
```
He tried to save the world with open-source software
A lot
```

Reply:
haaaa I am so sorry -- I wanted to fix the issue with the actual calculation quickly, and while I was testing I quickly tried the validation to make sure 0s are reflected -- I am sorry, I actually didn't mean to step on *your* toes here, I just wanted to quickly correct the bigger math, and this was added in.
See https://github.com/mooeypoo/buffycalculator/commit/804726e490771084c185d957aa993f5b8820d2c5

But you **did** contribute -- and I added an official thank you to the page, which I hope is okay: http://localhost/projects/buffycalculator/calculators/vampgrowth.html (see "Show updates and changes") -- do you want me to link anywhere?

Reply:
Haha, no worries at all! Thanks for the credit, and the link is perfect!

Let me know if you want to collaborate on any Buffyverse programming projects in the future - the world needs more people gentle of heart and nimble with a hard drive.